crates/diesel_models/src/payment_attempt.rs

```diff
@@ -4,35 +4,24 @@ use serde::{Deserialize, Serialize};
 use time::PrimitiveDateTime;
 
 use crate::enums::{self as storage_enums};
-#[cfg(all(any(feature = "v1", feature = "v2"), not(feature = "payment_v2")))]
+#[cfg(feature = "v1")]
 use crate::schema::payment_attempt;
-#[cfg(all(feature = "v2", feature = "payment_v2"))]
+#[cfg(feature = "v2")]
 use crate::schema_v2::payment_attempt;
 
-#[cfg(all(feature = "v2", feature = "payment_v2"))]
+#[cfg(feature = "v2")]
 #[derive(
     Clone, Debug, Eq, PartialEq, Identifiable, Queryable, Serialize, Deserialize, Selectable,
 )]
-#[diesel(table_name = payment_attempt, primary_key(attempt_id, merchant_id), check_for_backend(diesel::pg::Pg))]
+#[diesel(table_name = payment_attempt, primary_key(id), check_for_backend(diesel::pg::Pg))]
 pub struct PaymentAttempt {
-    pub payment_id: id_type::PaymentId,
+    pub payment_id: id_type::GlobalPaymentId,
     pub merchant_id: id_type::MerchantId,
-    pub attempt_id: String,
     pub status: storage_enums::AttemptStatus,
-    pub amount: MinorUnit,
-    pub currency: Option<storage_enums::Currency>,
-    pub save_to_locker: Option<bool>,
     pub connector: Option<String>,
     pub error_message: Option<String>,
-    pub offer_amount: Option<MinorUnit>,
     pub surcharge_amount: Option<MinorUnit>,
-    pub tax_amount: Option<MinorUnit>,
     pub payment_method_id: Option<String>,
```

Comment:
If we have `GlobalPaymentMethodId` available, we can change `String` to the `GlobalPaymentMethodId`.